### PR TITLE
mavlink: STATUSTEXT stream use perf count for missed messages

### PR DESCRIPTION
**Pull request by dagar**
- STATUSTEXT stream use perf count for missed messages instead of error message
 - don't send stale messages

### CODE REVIEW

**Review by julianoes — DISMISSED** [review]

File: src/modules/mavlink/streams/STATUSTEXT.hpp [hunk 1]
```diff
@@ -55,57 +57,69 @@ class MavlinkStreamStatustext : public MavlinkStream
...
+						msg.severity = mavlink_log.severity;
+						msg.chunk_seq = 0;
+						msg.id = _id++;
+						unsigned text_size;
 
-					if (chunk_size < max_chunk_size) {
-						memcpy(&msg.text[0], &text[0], chunk_size);
-						// pad with zeros
-						memset(&msg.text[0] + chunk_size, 0, max_chunk_size - chunk_size);
+						while ((text_size = strlen(text)) > 0) {
```
Comment (julianoes) [inline src/modules/mavlink/streams/STATUSTEXT.hpp:96]: What about using `chunk_size = strnlen(text, max_chunk_size)` here?

Comment (dagar) [reply src/modules/mavlink/streams/STATUSTEXT.hpp:96]: Sure

Comment (dagar) [reply src/modules/mavlink/streams/STATUSTEXT.hpp:96]: It looks like its still using both.

**Review by julianoes — APPROVED** [review]

**Comment by dagar** [comment]
> Hello.
> Seems like "don't send stale messages" breaks the purpose of [d9febbb](https://github.com/PX4/PX4-Autopilot/commit/d9febbbeccce0a78463273f9762c8e6e69cb4655) ("send status text only if connected to something that can actually handle it")
> This causes drone to drop STATUSTEXT messages if at least one heartbeat from GCS is lost at that moment.

We could widen the timeout a bit, but I don't think we should be sending all old queued messages regardless. 
Open a new issue to discuss?

**Comment by dagar** [comment]
> I think it's worth increasing either this hard-coded timeout or telemetry_status.HEARTBEAT_TIMEOUT_US, so mavlink does not enters "disconnected" state when just one heartbeat lost (that's how we've fixed it in our custom branch).

Could you followup in https://github.com/PX4/PX4-Autopilot/pull/19184 and let me know if that's sufficient in your experience?

**Pull request merged**